dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdSymbolFileUploadTaskTest.kt

```diff
@@ -0,0 +1,666 @@
+package com.datadog.gradle.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.datadog.gradle.plugin.internal.ApiKey
+import com.datadog.gradle.plugin.internal.ApiKeySource
+import com.datadog.gradle.plugin.internal.DdAppIdentifier
+import com.datadog.gradle.plugin.internal.Uploader
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.gradle.api.provider.Provider
+import org.gradle.internal.impldep.org.junit.Assume.assumeTrue
+import org.gradle.testfixtures.ProjectBuilder
+import org.json.JSONArray
+import org.json.JSONObject
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.api.io.TempDir
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.io.File
+import java.util.UUID
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DdSymbolFileUploadTaskTest {
```

Comment:
It is also worth adding some functional tests (see `DdAndroidGradlePluginFunctionalTest`), especially tests checking that if we call upload mapping task and also upload NDK symbol task for the same build, they will use the same build ID.